frontend/src/components/form/api/__tests__/ApiWrapper.spec.js

```diff
@@ -79,7 +69,17 @@ function createConfig (overrides) {
     default: '<input type="text" name="dummyField" id="dummyField" :value="props.localValue" />'
   }
 
-  return cloneDeep(Object.assign({ mocks, propsData, vuetify, stubs, scopedSlots }, overrides))
+  // Override the hal-json-vuex plugin
+  const localVue = createLocalVue()
+  Object.defineProperties(localVue.prototype, {
+    api: {
+      get () {
+        return apiStore
```

Comment:
Gibt es einen spezifischen Grund, wieso du hier den tatsächlichen apiStore hinein nimmst und nicht beim Mock geblieben bist?

Reply:
Ja, Jest hat etwas ähnliches gesagt wie "Cannot replace property 'api', this is usually caused by a plugin defining it as readonly". Habe keinen anderen Weg dazu gefunden, du darfst es aber gerne auch noch probieren.

Reply:
Das Problem ist, dass der apiStore in diesen Fällen eigentlich gar nicht geladen werden sollte (weil nicht benötigt). Wird aber trotzdem, weil unsere JS-Files unter /plugins als Nebeneffekte gleich den store installieren ohne den Aufruf von Vue.use. Wenn also irgendwo store/index.js importiert wird, wird auch gleich der Store aufgesetzt und ApiStore installiert.

Da reicht auch schon ein harmloses
```
import { formBaseComponents } from '@/plugins'
```

Weil der sucht formBaseComponents in plugins/index.js, und der lädt den store auch gleich gratis mit dazu 😅 

Hab mal einen Commit gepusht bei dem die Tests laufen ohne den Store zu importieren. Versuch mich mal noch, die Nebeneffekte zu entfernen.

Reply:
Nice, danke!

Ich sehe du hast noch einen Port geöffnet für Jest Debugging. PHPStorm unterstützt leider kein npm im Container, daher musste ich zum debuggen npm auch ausserhalb des Containers installieren. Aber um Breakpoints in den Vue-Komponenten zu setzen musste ich noch im package.json `collectCoverage` auf false setzen, sonst waren die Source Maps kaputt. Hast du diesen Effekt bei dir auch? Muss `collectCoverage` `true` sein damit die Coverage funktioniert oder könnte man das auch allgemein auf `false` lassen?

Reply:
Ne, funktioniert bei mir ohne. Wobei NPM debuggen immer ein wenig ein "Gefrickel" ist. Wobei breakpoints setzten schon etwas Glückssache ist.

CollectCoverage muss auf true sein um coverage zu sehen, ja.

Reply:
Okay dann lassen wirs so, und ich kann das Flag bei mir temporär auf `false` setzen wenn ich debuggen muss.